### ISSUE TITLE
py 3.12 -- test completion sourcing fails

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-cli#1441]
only on CI on ubuntu

```
============================= test session starts ==============================
platform linux -- Python 3.12.3, pytest-8.2.0, pluggy-1.5.0 -- /opt/hostedtoolcache/Python/3.12.3/x64/bin/python
cachedir: .pytest_cache
rootdir: /home/runner/work/dandi-cli/dandi-cli
configfile: tox.ini
plugins: mock-3.14.0, rerunfailures-14.0, timeout-2.3.1, cov-5.0.0
timeout: 300.0s
...
=================================== FAILURES ===================================
_______________________ test_shell_completion_sourceable _______________________
dandi/cli/tests/test_shell_completion.py:17: in test_shell_completion_sourceable
    subprocess.run(
/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/subprocess.py:571: in run
    raise CalledProcessError(retcode, process.args,
E   subprocess.CalledProcessError: Command '['bash', '-c', 'source <(dandi shell-completion)']' returned non-zero exit status 2.

---------- coverage: platform linux, python 3.12.3-final-0 -----------
Coverage XML written to file coverage.xml
```

but passes for me locally

```
❯ python -m pytest -s -v dandi/cli/tests/test_shell_completion.py
====================================================================================================== test session starts =======================================================================================================
platform linux -- Python 3.12.3, pytest-8.2.0, pluggy-1.5.0 -- /home/yoh/proj/dandi/dandi-cli-master/venvs/dev3.12/bin/python
cachedir: .pytest_cache
rootdir: /home/yoh/proj/dandi/dandi-cli-master
configfile: tox.ini
plugins: rerunfailures-14.0, mock-3.14.0, cov-5.0.0
collected 1 item                                                                                                                                                                                                                 

dandi/cli/tests/test_shell_completion.py::test_shell_completion_sourceable PASSED

====================================================================================================== slowest 10 durations ======================================================================================================
1.38s call     dandi/cli/tests/test_shell_completion.py::test_shell_completion_sourceable

(2 durations < 0.005s hidden.  Use -vv to show these durations.)
======================================================================================================= 1 passed in 1.42s ========================================================================================================
python -m pytest -s -v dandi/cli/tests/test_shell_completion.py  5.92s user 1.71s system 159% cpu 4.774 total

```

I guess would need an interactive session on CI to see what's up

**Comment by yarikoptic** [comment]
it is in the logs:

```
2024-05-13T06:06:23.9230867Z dandi/cli/tests/test_service_scripts.py::test_update_dandiset_from_doi[10.1038/s41467-023-37704-5-nature] PASSED
2024-05-13T06:06:25.6762661Z /dev/fd/63: line 1: syntax error near unexpected token `('
2024-05-13T06:06:25.6764801Z /dev/fd/63: line 1: `2024-05-13 06:06:24,036 [WARNING] DueCredit internal failure while running <function _get_active_due at 0x7fbe085b4860>: ModuleNotFoundError("No module named 'distutils'"). Please report to developers at https://github.com/duecredit/duecredit/issues (utils.py:194)'
2024-05-13T06:06:25.8632707Z dandi/cli/tests/test_shell_completion.py::test_shell_completion_sourceable FAILED
```
so we need

- make sure we do not spit out warnings while providing bash completion snippet (should close this issue)
- fix duecredit (just needs release)

**Comment by yarikoptic** [comment]
addressed by releasing duecredit 0.10.0 (had to upload to pypi "manually" since auto failed to trigger that due to e2big)
